src/protocols/secure_channel/CASESession.cpp

```diff
@@ -498,114 +461,103 @@ CHIP_ERROR CASESession::SendSigmaR2()
     VerifyOrExit(msg_salt.Alloc(saltlen), err = CHIP_ERROR_NO_MEMORY);
     VerifyOrExit(msg_rand.Alloc(kSigmaParamRandomNumberSize), err = CHIP_ERROR_NO_MEMORY);
 
-    // Step 1
     // Fill in the random value
     err = DRBG_get_bytes(msg_rand.Get(), kSigmaParamRandomNumberSize);
     SuccessOrExit(err);
 
-    // Step 3
-    // hardcoded to use a p256keypair
+    // Generate an ephemeral keypair
 #ifdef ENABLE_HSM_CASE_EPHERMAL_KEY
     mEphemeralKey.SetKeyId(CASE_EPHEMERAL_KEY);
 #endif
     err = mEphemeralKey.Initialize();
     SuccessOrExit(err);
 
-    // Step 4
+    // Generate a Shared Secret
     err = mEphemeralKey.ECDH_derive_secret(mRemotePubKey, mSharedSecret);
     SuccessOrExit(err);
 
-    err = ComputeIPK(mConnectionState.GetLocalKeyID(), mIPK, sizeof(mIPK));
-    SuccessOrExit(err);
-
-    // Step 5
     {
         MutableByteSpan saltSpan(msg_salt.Get(), saltlen);
-        err = ConstructSaltSigmaR2(ByteSpan(msg_rand.Get(), kSigmaParamRandomNumberSize), mEphemeralKey.Pubkey(), mIPK,
-                                   sizeof(mIPK), saltSpan);
+        err = ConstructSaltSigmaR2(ByteSpan(msg_rand.Get(), kSigmaParamRandomNumberSize), mEphemeralKey.Pubkey(),
+                                   ByteSpan(mIPK, sizeof(mIPK)), saltSpan);
         SuccessOrExit(err);
     }
 
     err = mHKDF.HKDF_SHA256(mSharedSecret, mSharedSecret.Length(), msg_salt.Get(), saltlen, kKDFSR2Info, kKDFInfoLength, sr2k,
                             kAEADKeySize);
```

Comment:
`kAEADKeySize` should be deleted from this file and every mention of it should be the canonical `CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BITS` (which I now realize is missing from CHIPCryptoPAL.h :( 

This could be left as a TODO where kAEADKeySize is defined.